papers/MSM/H23 - Notes on MSMs with Precomputation.md

```
@@ -0,0 +1,198 @@
+so we can view $a_i$'s as have this many bits.
+
+[^Bandersnatch]: "Bandersnatch: a fast elliptic curve built over the
+BLS12-381 scalar field" by Simon Masson, Antonio Sanso and Zhenfei Zhang https://eprint.iacr.org/2021/1152.pdf
+
+Due to the fact that the $P_1,\ldots,P_{256}$ are fixed, we may store large pre-computed tables of curve points in memory and try to use those to then speed up the computation of $a_1P_1 + \ldots + a_{256}P_{256}$.
+
+## Naive algorithm
+
+A somewhat naive way is to binary decompose each $a_i$ into bits $a_i = \sum_i a_{i,j} 2^j$ with $a_{i,j}\in\{0,1\}$. This means we decompose the exponents into a matrix of bits.
+
+![](./images/H23%20-%20Notes%20on%20MSMs%20with%20Precomputation/H23%20-%20Fig%201%20-%20Naive%20algorithm%20binary%20decomposition.png)
```

markdownlint-cli2 (0.22.1)

[warning] Images should have alternate text (alt text)

(MD045, no-alt-text)